container/options.go

```diff
@@ -216,6 +218,19 @@ func SplitPercent(p int) SplitOption {
 	})
 }
 
+// SplitFixed sets the heights of the top and the bottom containers.
+// The first container will be fixed and the second one will fill the rest of the space.
+// Allows values greater or equal to zero.
+func SplitFixed(cells int) SplitOption {
+	return splitOption(func(opts *options) error {
+		if cells < 0 {
+			return fmt.Errorf("invalid fixed value %d, must be in range %d <= cells", cells, 0)
```

Comment:
Now that we have two ways of specifying split size, we should validate that only one of them is ever set. I.e. return an error if both are set.

Such validation can be added into this function:

https://github.com/mum4k/termdash/blob/0e5242ca6f8386889a303c67b58a25f98aed75b6/container/options.go#L41-L61

Reply:
Split the `validateOptions` into two sub functions -- `validateIds` and `validateSplits`.
`validateIds` checks that no duplicate ids exists, and `validateSplits` check that SplitFixed and SplitPercent are not specified at the same time.

Reply:
It is fine to split these into two functions, more focused functions are easier to digest. We should however consider the cost of the preOrder traversal. The containers are organized in a binary tree and now each of the validation functions traverses from the root container to all the leaves, visiting and validating each container.

While we probably aren't going to feel the performance impact immediately, it might be a low hanging fruit to perform the traversal just once and call the two validation functions on each visited container.

Do you think this is something worth fixing?

Reply:
Makes sense. But yeah, as you said, no big impact atm, although fixed it anyways.